deps/rabbit/apps/rabbitmq_prelaunch/src/rabbit_prelaunch_dist.erl

```diff
@@ -49,6 +52,19 @@ do_setup(#{nodename := Node, nodename_type := NameType}) ->
             {ok, _} = net_kernel:start([Node, NameType]),
             ok
     end,
+
+    %% Override the Erlang cookie if one was set in the environment.
+    case maps:get(erlang_cookie, Origins, default) of
+        environment ->
+            ?LOG_DEBUG(
```

Comment:
Should this be a warning? It can be a painful thing to troubleshoot, since `rabbitmq-diagnostics` and friends depend on a cookie match.